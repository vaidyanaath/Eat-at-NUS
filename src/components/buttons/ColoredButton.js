import styled from 'styled-components/native';
import { colors } from '../../assets/colors';


export const ColoredButton = styled.TouchableOpacity`
    padding: 10px;
    margin: 5px;
    border-radius: 10px;
    align-items: center;
    justify-content: center;
    align-self: center;
    background-color: ${colors.primary};
    shadowColor: #000;
    shadowOffset: 0 1px;
    shadowOpacity: 0.22;
    shadowRadius: 2.22px;
    elevation: 3;
`;
