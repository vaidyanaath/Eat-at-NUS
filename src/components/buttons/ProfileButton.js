import React from 'react';
import { TouchableOpacity, StyleSheet, Image } from 'react-native';

// import colors
import { colors } from '../../assets/colors';

// import auth
import { auth } from '../../firebase/config';
import { signOut } from 'firebase/auth';

const handleProfileButton = () => {
    console.log("profile button pressed!");
    signOut(auth).then(() => {
        // Sign-out successful.
        console.log("signed out");
    }).catch((error) => {
        console.log(error);
    });
}

export const ProfileButton = ({ source }) => (
    <TouchableOpacity onPress={handleProfileButton} backgroundColor={colors.black} style={styles.profilePic}>
        <Image 
            style={styles.profilePic} 
            source={source}
        />
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    profilePic: {
        width: 40,
        height: 40,
        borderRadius: 20,
    }
})
