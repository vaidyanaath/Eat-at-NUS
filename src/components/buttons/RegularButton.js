import styled from 'styled-components/native';
import { colors } from '../../assets/colors';


export const RegularButton = styled.TouchableOpacity`
    padding: 10px;
    margin: 5px;
    border-radius: 10px;
    align-items: center;
    justify-content: center;
    align-self: center;
    background-color: ${colors.bg};
`;
