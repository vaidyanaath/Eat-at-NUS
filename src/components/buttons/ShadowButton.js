import styled from 'styled-components/native';

export const ShadowButton = styled.TouchableOpacity`
    padding: 10px;
    margin: 5px;
    border-radius: 10px;
    align-items: center;
    justify-content: center;
    align-self: center;
    shadow-color: #000;
    shadow-offset: { width: 5, height: 5 };
    shadow-opacity: 0.22;
    shadow-radius: 2.22;
    elevation: 3;
`;
