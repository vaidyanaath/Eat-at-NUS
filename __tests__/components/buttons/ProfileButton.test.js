import React from 'react';
import renderer from 'react-test-renderer';

import { ProfileButton } from '../../../src/components/buttons/ProfileButton';

describe('<ProfileButton />', () => {
    
    it('Renders correctly', () => {
        const tree = renderer.create(<ProfileButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});