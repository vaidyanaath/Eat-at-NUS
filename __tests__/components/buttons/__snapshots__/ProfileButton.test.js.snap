// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileButton /> Renders correectly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 20,
      "height": 40,
      "opacity": 1,
      "width": 40,
    }
  }
>
  <Image
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      }
    }
  />
</View>
`;
